registration/api/v2/views.py

```diff
@@ -185,6 +188,54 @@ def perform_update(self, serializer):
     def partial_update(self, request, *args, **kwargs):
         return super(OpenEdXInstanceConfigViewSet, self).partial_update(request, *args, **kwargs)
 
+    @swagger_auto_schema(
+        request_body=ThemeSchemaSerializerGenerator,
+        responses={**VALIDATION_RESPONSE, 200: ThemeSchemaSerializerGenerator, },
+    )
+    @action(detail=True, methods=["patch"])
+    def theme_config(self, request, pk=None):
+        """
+        Partial update for theme configuration
+
+        This is a custom handler to partially update theme fields.
+        """
+        application = self.get_object()
+
+        # Sanitize inputs
+        serializer = ThemeSchemaSerializerGenerator(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        # Merges the current aplication theme draft with the values modified by
+        # the user and performs validation.
+        # If the key is empty (""), the user reverted the field to the default
+        # value, and the key needs to be erased.
+        merged_theme = {
+            key: value for key, value in {
+                **application.draft_theme_config, **serializer.validated_data
+            }.items() if value != ""
+        }
+        # To make sure the required values are always available, merge dict with
+        # default values dictionary.
+        # This API only changes values for the v1 schema, so it's hardcoded here
+        safe_merged_theme = {
+            'version': 1,
+            **DEFAULT_THEME,
+            **merged_theme
+        }
```

Comment:
>Could you explain why this is needed?
This code is needed because the code above erases default values and removes the keys if the user empties a field. Since this would break the provisioning of an instance I overlay the default theme with the changes sent by the user, so if a required variable is erased, it gets set to the default value.

>Won't application.draft_theme_config always contain all attributes in DEFAULT_THEME and 'version' == 1?
No for all `DEFAULT_THEME` variables for the reason explained above. But yes., I can remove the `'version' == 1` since it's already present in the default theme.